eventdiscovery-sdk/src/main/java/com/schedjoules/eventdiscovery/eventdetails/BasicEventDetails.java

```diff
@@ -55,54 +63,48 @@ public BasicEventDetails(@NonNull Event event)
     @Override
     public void show(@NonNull final Activity activity)
     {
-        new Thread(new StartEventDetailsRunnable(activity, mEvent)).start();
-    }
-
-
-    /**
-     * A {@link Runnable} that starts the details activity in the background after trying to load any actions from the cache.
-     */
-    private final static class StartEventDetailsRunnable implements Runnable
-    {
-        private final Activity mActivity;
-        private final Event mEvent;
-
+        final OptionalArgument<MicroFragmentHost> host = new OptionalArgument<>(Keys.MICRO_FRAGMENT_HOST, activity);
 
-        private StartEventDetailsRunnable(@NonNull Activity activity, @NonNull Event event)
+        if (!host.isPresent())
```

Comment:
It was for being able to signal that `BasicEventDetails` can only be used from Activtiy with MFHost. But you are right, it's unnecessary, so I removed it and use it as mandatory with new convenience class `ActivityMicroFragmentHost implements Lazy<MicroFragmentHosts>`.